### PR TITLE
adding some exercises

### PR DESCRIPTION
**Pull request by tarquin-the-brave**
Hopefully these give a some more body to the earlier chapters.
I think the antistrings exercise is in an OK place.  It would be interesting to try with only the knowledge in the book up to that chapter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +41,66 @@ Write a function to generate the fibonacci numbers
...
+## Higher Order Functions
+
+### Exercise (primes)
+
+Write functions that produce:
+* The 1st n primes
+* The nth prime
+* The first prime above a given integer
+
+Attempt the functions after reading **Maps and Filters**.
```
Comment (rlupton20) [inline README.md]: Do you mean the prime functions above?

Comment (tarquin-the-brave) [reply README.md]: Ah yeah, I'll make that a bit more clear

File: README.md [hunk 2]
```diff
@@ -34,6 +41,66 @@ Write a function to generate the fibonacci numbers
...
+Load this module into ghci.
+
+## Making our own Types and Typeclasses
+
+### Exercise (anti-strings)
+
+In physics there re particles and antiparticles.  When a particle meets it's antiparticle counterpart they annihilate to nothing!
+Introducing anti-strings! E.g:
+```
+"disfunctional" ++ Antistring "fun" => "disctional"
```
Comment (rlupton20) [inline README.md]: Eugh - do you really want to look inside the string (as opposed to annihilating the right-hand-side in the above example? What happens if you don't find "fun" inside the string you add to?)

Also `(++) :: [a] -> [a] -> [a]`, whereas you need something else here. Would be worth writing down a few more examples of expected behaviour, and checking the exercise is possible in the framework you lay out (is it?).

As a general principle, it's worth noting that Haskell's typeclasses have a strong algebraic flavour. Strings form something called a `Monoid` under `(++)` - it looks like you are trying to make a `Group` - or rather that would be what I would expect, I guess.

Maybe actually your above example does make sense, I'm just misled by the fact that `(++)` has quite a principled behaviour.

File: README.md [hunk 3]
```diff
@@ -11,12 +13,17 @@
...
-  - if i is (only) divisible by 3 then x is the string `"Fizz"`; 
+  - if i is (only) divisible by 3 then x is the string `"Fizz"`;
   - if i is (only) divisible by 5 then x is the string `"Buzz"`;
   - if i is divisible by 3 and 5 then x is the string `"FizzBuzz"`;
   - otherwise x is the string `""`;
 `fizzBuzz` should be an infinite list. You may define it any way you please.
 
+### Exercise (intermediate)
+
+Write a function that implements an aproximation to sqrt for integers.
```
Comment (rlupton20) [inline README.md]: I'm assuming you mean `floor . sqrt`, in essence. Would be worth indicating that (examples?), without giving the types away (my assumption here is the function has type `Integer -> Maybe Integer`).

**Review by rlupton20 — COMMENTED** [review]
Looks reasonable enough!

I've left some comments where I was unsure what you meant, or I thought there might be some issues, but it's generally fine.

Run a spellcheck over it too if possible - if that's arse I can fix up typose later.

File: README.md (hunk 2 above)
Comment (tarquin-the-brave) [reply README.md]: Yeah I haven't really thought this through properly yet.  I'll make it less specific and open-ended. Perhaps put it under a later chapter.

File: README.md (hunk 3 above)
Comment (tarquin-the-brave) [reply README.md]: So my solution was:
```haskell
sqrtInt = \n -> last [x | x <- [1..n], x*x <= n]
```
I didn't know about `floor`.  I'll think about how to phrase the question  to basically say "don't just do `floor . sqrt`"

**Comment by tarquin-the-brave** [comment]
I'll find a good spellchecker for vim.

**Comment by rlupton20** [comment]
Ta (for fixing up my spellings too!).

I remember your integer square root function now - I didn't know `floor` and `sqrt` were available in `Prelude`, but I'd have had a go and got lucky I guess :P.

**Pull request merged**